analytics/lib/counts.py

```diff
@@ -82,6 +94,23 @@ def last_successful_fill(self) -> datetime | None:
             return fillstate.end_time
         return fillstate.end_time - self.time_increment
 
+    def current_month_accumulated_count(self) -> int:
+        # Sum is generated across the server and not for a realm.
+        now = timezone_now()
+        start_of_month = datetime(now.year, now.month, 1, tzinfo=timezone.utc)
+
+        # We just want to check we are not using BaseCount, otherwise all
+        # `output_table` have `objects` property.
+        assert self.data_collector.output_table != BaseCount
+        result = self.data_collector.output_table.objects.filter(  # type: ignore[attr-defined] # see above
+            property=self.property,
+            end_time__gte=start_of_month,
+            end_time__lte=end_of_month(now),
```

Comment:
Can't we just use `__lt=` and the first day of the next month, rather than constructing this `end_of_month` variable? I think `datetime` supports a `timedelta(months=1)` you can add to the `start_of_month` window.

Reply:
Yeah, looks like I overengineered it.